site/source/tutorials/working_with_virtual_dom.md

```diff
@@ -0,0 +1,106 @@
+
+{% aside 'Virtual DOM projectors' %}
+A virtual DOM renders content to the web page using the DOM just like any other web application. A **projector** is a component of the virtual DOM that synchronizes the virtual DOM nodes and the DOM.
+{% endaside %}
+
+Consider an example where a reuseable page title component is required. This widget must be able to vary the content inside of an `h1` element. Using the DOM, we might end up with a method like this:
+
+```ts
+  updateTitle(title: string): void {
+    this.pageTitle.innerHTML = title;
+  }
+```
+
+This code requires the widget instance to track the DOM element that contains the title text and set the correct property whenever it is called. In contrast, the same thing is done in Dojo 2 with this method:
+
+```ts
+  render(): DNode {
+    return v('h1', { class: 'title' }, [ this.title ]);
+  }
+```
+
+Notice how similar this is to the original statement that returned static content? Since the projector automatically determines the differences, the application does not have to track changes - it simply describes the elements as they should exist. This greatly reduces the API required to describe the content of a web page.
+
+### Efficiently manage DOM changes
+
+Almost all interactions with a web application cause a visual update of some kind. Many of these updates require the page's DOM structure to be changed by adding, deleting, or moving elements. When this occurs, the web browser must recalculate the position of every element and then re-render them. When performed too often, this process can degrade the performance and responsiveness of the application. Consider this example that is adding a series of items to a list:
+
+```ts
+  const ul = document.getElementById('theList');
+  for (let i = 0; i < 5; i++) {
+    const li = document.createElement('li');
+    li.value = i;
+    li.innerHTML = i;
+    ul.appendChild(li);
+  }
+```
+
+This code will cause the browser to recalculate the flow of the page's elements five times - once per item added to the list. There are ways to eliminate this, using `DocumentFragments` for example, but they often require additional code to be written that obscures the intent of the function. Since an application that uses a virtual DOM doesn't concern itself with intermediate changes, only the final result is required to be generated:
+
+```ts
+  const values = [0, 1, 2, 3, 4];
+
+  v('ul', values.map(value => v('li', [ value ]));
+```
+
+This code returns the `ul` element with the required children in place. When the virtual DOM renders this update, it will determine how to most efficiently render this element.
+
+The virtual DOM achieves its performance improvements by limiting the number of times the DOM is updated and, when it is updated, ensuring that the update is performed as efficiently as possible.
```

Comment:
"as efficiently as possible" it feels like we've said this too much without saying how. We could say something like, for example, if creation of a DocumentFragment is considered most efficient by the Virtual DOM system, then it will internally create and insert content in that manner.  ? or maybe what we have later is sufficient, but at this point I want to know how it does that.